CMakeLists.txt

```diff
@@ -75,24 +75,27 @@ include(CTest)
 if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
     #set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=address,memory,undefined")
     #set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Weverything")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wpedantic")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wpedantic -Wshadow")
     string(CONCAT CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} "
-                  "-Wno-weak-vtables -Wno-padded -Wno-switch-enum "
-                  "-Wno-undefined-func-template -Wno-c++98-compat-pedantic")
+            "-Wno-unknown-pragmas")
     # silence HDF5 (FIXME: -isystem should be enough)
     string(CONCAT CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} "
                   "-Wno-reserved-id-macro -Wno-deprecated -Wno-old-style-cast "
                   "-Wno-cpp")
     # older clangs: silence unknown warnings
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unknown-warning-option")
 elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wpedantic")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wpedantic -Wshadow")
+    string(CONCAT CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} "
+            "-Wno-unknown-pragmas")
     # silence BOOST filesystem
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-cpp")
 elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
     # Warning C4503: "decorated name length exceeded, name was truncated"
     # Symbols longer than 4096 chars are truncated
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -wd4503")
+else ()
```

Comment:
Usually we would not place warnings in cmakelists anyways since they are not required for a build. We just have to for now since we have not written cmake Toolchain files for Travis yet.
Anyway, don't warn please ;)

Reply:
So would you prefer a plain old ```message()``` or no ```else``` at all?